span_sink.go

```diff
@@ -398,10 +407,91 @@ func (b *blackholeSpanSink) Start(*trace.Client) error {
 	return nil
 }
 
-func (b *blackholeSpanSink) Ingest(ssf.SSFSpan) error {
+func (b *blackholeSpanSink) Ingest(*ssf.SSFSpan) error {
 	return nil
 }
 
 func (b *blackholeSpanSink) Flush() {
 	return
 }
+
+type metricExtractionSink struct {
+	workers                []*Worker
+	indicatorSpanTimerName string
+}
+
+var _ sinks.SpanSink = &metricExtractionSink{}
+
+func NewMetricExtractionSink(mw []*Worker, timerName string) (*metricExtractionSink, error) {
+	return &metricExtractionSink{mw, timerName}, nil
+}
+
+// Name returns "metric_extraction".
+func (m metricExtractionSink) Name() string {
+	return "metric_extraction"
+}
+
+// Start is a no-op.
+func (m metricExtractionSink) Start(*trace.Client) error {
+	return nil
+}
+
+func (m *metricExtractionSink) sendMetrics(metrics []samplers.UDPMetric) {
+	for _, metric := range metrics {
+		m.workers[metric.Digest%uint32(len(m.workers))].PacketChan <- metric
```

Comment:
I think this code is a failure of interface on our part. As we discussed in Slack I think we should come up with something better here in the future. Not your problem right now though!

Reply:
Agreed - there's a lot going on in this that could go wrong (: